client/landing/jetpack-cloud/sections/scan/controller.js

```diff
@@ -2,22 +2,56 @@
  * External dependencies
  */
 import React from 'react';
+import { noop } from 'lodash';
 
 /**
  * Internal dependencies
  */
 import ScanPage from './main';
 import ScanHistoryPage from './history';
+import ScanUpsellPage from './upsell';
+import { getSelectedSiteId } from 'state/ui/selectors';
+import { getSitePurchases } from 'state/purchases/selectors';
+import { JETPACK_SCAN_PRODUCTS, PRODUCT_JETPACK_SCAN } from 'lib/products-values/constants';
+import { getPlan, planHasFeature } from 'lib/plans';
+import { fetchSitePurchases } from 'state/purchases/actions';
+import { makeLayout, render as clientRender } from 'controller';
+
+export function showUpsellIfNoScan( context, next ) {
+	const getState = context.store.getState;
+	const dispatch = context.store.dispatch;
+	const siteId = getSelectedSiteId( getState() );
+	let purchases = getSitePurchases( getState(), siteId );
+
+	const showUpsellOrNext = () => {
+		const hasScan = purchases.find(
+			purchase =>
+				JETPACK_SCAN_PRODUCTS.includes( purchase.productSlug ) ||
```

Comment:
I think I have seeing the same check in other places – checking if a user has a product by looking at purchases and plans. Do you think creating a selector would be a good idea? I'm referring to this piece of code:
```js
JETPACK_SCAN_PRODUCTS.includes( purchase.productSlug ) ||
( getPlan( purchase.productSlug ) &&
	planHasFeature( purchase.productSlug, PRODUCT_JETPACK_SCAN ) )
```

Reply:
@rcanepa - We're discussing creating a specific API endpoint to query what the current state of scan is for a site, so I feel it's not worth the time to create a selector that queries purchases like this. Once we have the API in place we can deprecate this approach in favor of the new selector.

Reply:
Cool. That makes much more sense!